ApollosProject/apollos-prototype
# Push Provider/Consumer/Context

## DESCRIPTION

### What does this PR do, or why is it needed?

Handles the state where the user has already been prompted for notifications and has declined. I previously tried [this](https://github.com/ApollosProject/apollos-prototype/pull/1243), but found that I could not enable notifications, if a user had ever declined them (makes sense from a user protection standpoint, but is very obnoxious). I also tried using listeners, but to no avail.

Now instead, I disable the notifications button to show the choice has already been made; I disable the"skip" button and enable the "finish" button 

This version of Push handling uses Provider/Consumer/Context in order to better handle push settings. It should address the problems [here](https://github.com/ApollosProject/apollos-prototype/pull/1246)

### What design trade-offs/decisions were made?

- Not sure this solution is great design-wise, but it seemed to look alright and make sense to me.
- There is a loading state for the button, but the change is so quick it's hard to see
- Changing your settings through the settings menu will not update the UI;  this is because RN does not have access to permission listeners. If there is a way to track background/foreground, we can fix this.

### How do I test this PR?

1. Delete the app from your sim
1. In onboarding tap Enable Notifications and reject the hard ask
3. In onboarding the state of this screen should change
4. Repeat the previous steps, but accept the hard ask

### What automated tests did you write?

## SCREENSHOTS

<img width="570" alt="Screen Shot 2019-11-21 at 3 47 54 PM" src="https://user-images.githubusercontent.com/29125070/69375606-bd4f8180-0c76-11ea-9f59-92f1c3741543.png">

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
